addons/account/models/account_move.py

```diff
@@ -3367,6 +3367,8 @@ def create(self, vals_list):
                         taxes,
                         move.type,
                     ))
+                    if move.type in ['out_invoice','out_refund'] and vals.get('product_id') and not vals.get('name'):
```

Comment:
```suggestion
                    if not vals.get('name') and move.type in ['out_invoice','out_refund'] and vals.get('product_id'):
```
This will be fast and more obvious

Reply:
Thanks